app/components/App/App.js

```diff
@@ -0,0 +1,207 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { autobind } from 'core-decorators';
+import { BrowserRouter, Route, Switch } from 'react-router-dom';
+import findPath from '../../utils/BFS';
+import Game from '../Game/Game';
+// const Settings = require('./Settings');
+// const Results = require('./Results');
+
+function resetBoard(lengthX, lengthY) {
+  console.log('creating board');
+
+  const board = [];
+  for (let i = 0; i < lengthY; i += 1) {
+    board[i] = [];
+    for (let j = 0; j < lengthX; j += 1) {
+      board[i][j] = {
+        isAllowedToBeSteppedOn: true,
+        numberOfStepInPath: -1,
+      };
+    }
+  }
+
+  return board;
+}
+
+function cloneBoard(board) {
+  return (board.map(row => (row.map(cell => ({ ...cell, })))));
+}
+
+@autobind
```

Comment:
I prohibit my team members from binding at the class level. We `autobind` only methods which reference `this` and are called in a context which would modify binding (thus requiring binding).

Reply:
done